engine/src/main/java/org/hibernate/validator/internal/engine/ConfigurationImpl.java

```diff
@@ -494,23 +498,28 @@ private void parseValidationXml() {
 			);
 			applyXmlSettings( xmlParameters );
 		}
-
-		for ( ValueExtractorDescriptor valueExtractorDescriptor : valueExtractorDescriptors.values() ) {
-			validationBootstrapParameters.addValueExtractorDescriptor( valueExtractorDescriptor );
-		}
 	}
 
 	private void loadValueExtractorsFromServiceLoader() {
-		@SuppressWarnings("rawtypes")
-		ServiceLoader<ValueExtractor> loader = ServiceLoader.load(
-				ValueExtractor.class,
-				externalClassLoader != null ? externalClassLoader : getClass().getClassLoader()
-		);
+		List<ValueExtractor<?>> valueExtractors;
+
+		if ( externalClassLoader != null ) {
+			final ClassLoader originalContextClassLoader = run( GetClassLoader.fromContext() );
```

Comment:
Instead of setting the TCCL, I'd suggest to pass the CL as a parameter to GetValueExtractorsFromServiceLoader#action()`. We'd either pass the external CL or the current TCCL. Behavior should be the same, but a bit less convoluted.